lib/client-assets.php

```diff
@@ -704,13 +704,13 @@ function gutenberg_extend_settings_link_color( $settings ) {
 add_filter( 'block_editor_settings', 'gutenberg_extend_settings_link_color' );
 
 /*
- * Register default patterns if not registered in Core already.
+ * Register default patterns, potentially overriding ones that were already registered in Core.
  *
- * This can be removed when plugin support requires WordPress 5.5.0+.
+ * This can be removed when plugin support requires WordPress 5.5.0+, and patterns have been synced back to Core.
  *
  * @see https://core.trac.wordpress.org/ticket/50550
  */
-if ( class_exists( 'WP_Block_Patterns_Registry' ) && ! WP_Block_Patterns_Registry::get_instance()->is_registered( 'text-two-columns' ) ) {
+if ( class_exists( 'WP_Block_Patterns_Registry' ) ) {
 	register_block_pattern( 'core/text-two-columns', gutenberg_load_block_pattern( 'text-two-columns' ) );
```

Comment:
Implementation note: We don't have to unregister explicitly before re-registering, since the block pattern registry [simply overrides the entry for the pattern with the same name](https://github.com/WordPress/gutenberg/blob/0ac90749e99efe1871dd017fadad25e0ef80cd02/lib/class-wp-block-patterns-registry.php#L31-L63) (it's basically an assoc array).

Reply:
That should probably be fixed IMO. I think we should throw an error. (better as a trac ticket now though)

Reply:
It's separate from this PR though

Reply:
The issue in the tests is probably a "timing" issue. This may need to be run on "init".

Reply:
d638aef